driver: flash: Add Set/ Get write protect function

benson0715 opened the pull request:
Add Set_WP function to set SPI flash WP line to low 
Add Get_WP function to obtain status of the SPI flash WP line

keith-zephyr commented on drivers/flash/flash_realtek_rts5912.c:
```diff
@@ -427,20 +427,44 @@ static int flash_read_sr2(const struct device *dev, uint8_t *val)
 #endif
 
+#ifdef CONFIG_FLASH_EX_OP_ENABLED
+#define SPIC_BASE 0x40010200ul
```
This base address is already captured by the driver config structure.

keith-zephyr commented on drivers/flash/flash_realtek_rts5912.c:
```diff
@@ -427,20 +427,44 @@ static int flash_read_sr2(const struct device *dev, uint8_t *val)
+#define SPIC_BASE 0x40010200ul
+static int flash_set_wp(const struct device *dev, uint8_t *val, uint8_t cnt)
+{
+	volatile uint32_t *reg = (volatile uint32_t *)(SPIC_BASE + 0x110u);
```
You need to define register 0x110 in the `struct reg_spic_reg` structure.

```suggestion
	const struct flash_rts5912_dev_config *config = dev->config;
	volatile struct reg_spic_reg *spic_reg = config->regs;
```

benson0715 replied:
Done.

keith-zephyr commented on drivers/flash/flash_realtek_rts5912.c:
```diff
@@ -427,20 +427,44 @@ static int flash_read_sr2(const struct device *dev, uint8_t *val)
+
+static int flash_get_wp(const struct device *dev, uint8_t *val)
+{
+	volatile uint32_t *reg = (volatile uint32_t *)(SPIC_BASE + 0x110u);
```
Same here - use `struct reg_spic_reg` to access the register at offset 0x110.

benson0715 replied:
Done.

keith-zephyr commented on drivers/flash/flash_realtek_rts5912.c:
```diff
@@ -427,20 +427,44 @@ static int flash_read_sr2(const struct device *dev, uint8_t *val)
 #endif
 
+#ifdef CONFIG_FLASH_EX_OP_ENABLED
+static int flash_set_wp(const struct device *dev, uint8_t *val, uint8_t cnt)
```
`cnt` parameter is not used.

benson0715 replied:
Done.

benson0715 committed:
driver: flash: Add Set/ Get write protect function

Add Set_WP function to set SPI flash WP line to low
Add Get_WP function to obtain status of the SPI flash WP line

Signed-off-by: Benson Huang <benson7633769@gmail.com>

kartben commented on drivers/flash/flash_realtek_rts5912.c:
```diff
@@ -751,6 +775,12 @@ static int flash_rts5912_ex_op(const struct device *dev, uint16_t opcode, const
 	case FLASH_RTS5912_EX_OP_RD_SR2:
 		ret = flash_read_sr2(dev, (uint8_t *)in);
 		break;
+	case FLASH_RTS5912_EX_OP_SET_WP:
+		ret = flash_set_wp(dev, (uint8_t *)out);
+		break;
+	case FLASH_RTS5912_EX_OP_GET_WP:
+		ret = flash_get_wp(dev, (uint8_t *)in);
+		break;
```
in and out are backwards, I think (set should use "in", get should put data in "out")
